### PR TITLE
Check for serviceWorker support before attempting to add event listeners

### LINKED ISSUE #10: Check for serviceworker support
Safari 10 doesn't support serviceworkers, so line 111 causes the following error:

    TypeError: undefined is not an object (evaluating 'navigator.serviceWorker.addEventListener')

Since Safari still supports appcache, can jakecache let that take over as a fallback?

### PR DESCRIPTION
**Pull request by ejb**
This fixes #10, and is the last of my three pull requests.

Thanks for responding to my issues so quickly! OSS at its best :)

### CODE REVIEW

File: jakecache.js [hunk 1]
```diff
@@ -106,46 +106,48 @@ class JakeCache extends PolyfilledEventTarget {
...
-        case 'obsolete':
-          this[_status] = this.OBSOLETE
-          this.dispatchEvent(new CustomEvent('obsolete'))
-          break
-        case 'error':
-          this.dispatchEvent(new ErrorEvent('error', event.data))
-          break
-      }
-    })
+    if (this.pathname && 'serviceWorker' in navigator) {
```
Comment (kenchris) [inline jakecache.js]: Woudln't an early return not have made more sense? It also avoid the extra indentation

**Comment by ejb** [comment]
Oh, good idea. I can't make the change right now but will do soonish.

**Comment by ejb** [comment]
I've changed this to an early return, as suggested.

**Comment by kenchris** [comment]
OK cool, in the future you can amend the former commit, as this is mostly reverting it

**Comment by ejb** [comment]
Thanks for merging! Could you update the npm release please?

**Pull request merged**